src/operator/tensor/matrix_op-inl.h

```diff
@@ -285,8 +322,9 @@ void TransposeImpl(RunContext ctx,
      case 2: {
       mshadow::Tensor<xpu, 2, DType> in = src.FlatTo2D<xpu, DType>(s);
       mshadow::Tensor<xpu, 2, DType> out = ret.FlatTo2D<xpu, DType>(s);
+
       if (axes[0] == 1 && axes[1] == 0) {
-        out = in.T();
+        Transpose2D<xpu, DType>(in.dptr_, out.dptr_, in.shape_[0], in.shape_[1]);
```

Comment:
Why not just pass in in.shape_?

Reply:
Can work either way, but passing 2 shapes as params or as combined.